pkg/domain/entities/images.go

```diff
@@ -306,6 +306,24 @@ type ImageSaveOptions struct {
 	Quiet bool
 }
 
+// ImageScpOptions provide options for securely copying images to podman remote
+type ImageScpOptions struct {
+	// DestImageName is the image the user desires to load on their machine or remote machine
+	DestImageName string
+	// Tag allows for a new image to be created under the given name
+	Tag string
```

Comment:
Nothing sets this AFAICS.

Reply:
right now, is used to tell the user they cant rename the image on the loaded machine, I think this is good to avoid errors...

Reply:
OK, so the to values are actually `SourceImageName`, `DestImageName[IfDifferentFromSource]`? Maybe name them something like that.

---

It would make good sense to define a data structure that can represent save-under-different-name requests, and for the execution part to reject such requests; but that only works if the CLI parser actually parses such requests correctly, which is AFAICS not the case right now.

Reply:
but the image is always destination image since we cannot rename. Once we add renaming I will of course switch to multiple image names but right now imageSource and imageDest will just be copies of eachother, no point really.

Reply:
I mean, either write the code with a pervasive assumption that we are not renaming, and only have a single field for the source name (with `parseArgs` not supporting syntax for rename, or recognizing it and rejecting it); or write the code with a pervasive assumption that renames could happen, have two fields for the source/destination name (with `parseArgs` supporting full syntax for rename, and the execution rejecting it).

Now it’s an inconsistent mix of the two, with two variables as if renames were supported, but their names and uses strongly assuming no renames. So, in a hypothetical future where we permanently decide whether to support renames, _both_ `parseArgs` and the execution would need to be modified and cleaned up to be consistent.

Reply:
Ok so I will add `DestImageName` and `SourceImageName` but `DestImageName` should be unused for now since we are actually only using 1 name.

Reply:
added rejection for image renaming in all areas